#158174586 chore(api-tests) Fix API tests

magicmarie opened the pull request:
#### What does this PR do?
- Fixes unit tests in  hc/api/tests

#### Description
The app hc/api has its unit tests in hc/api/tests. Some of these tests were incomplete and needed fixing and some asserts added.

#### How should this be manually tested?
After cloning the repo, run the following commands
- python manage.py test

#### What are the relevant pivotal tracker stories?
[#158174586](https://www.pivotaltracker.com/n/projects/2177206)

magicmarie committed:
chore(api-tests) Fix API tests

This commit fixes all tests in hc.api.tests

Fix tests for sending alerts, notifications, checking models

[Delivers #158549842]

magicmarie committed:
chore(api-tests) Fix API tests

This commit fixes all tests in hc.api.tests

Fix tests for sending alerts, notifications, checking models

[Delivers #158549842]

magicmarie committed:
chore(heroku-deployment)Deploy the application to heroku
This commit  deploys the application to heroku

[Delivers #158549842]

matthewwacha commented on hc/api/tests/test_admin.py:
```diff
@@ -8,7 +8,10 @@ def setUp(self):
-        ### Set Alice to be staff and superuser and save her :)
+        #### Set Alice to be staff and superuser
+        self.alice.staff = True
+        self.alice.supersu = True
```
You made a typo on line[13]  ```self.alice.superuser = True```

EdwinKato replied:
Nice attempt, this is however not doing what is required. Hint! Check the database schemas and rectify this

mutebironald replied:
we will rectify this. Thanks

matthewwacha commented on hc/api/tests/test_check_model.py:
```diff
@@ -5,29 +5,30 @@
 from hc.api.models import Check
 
 
+check = Check()
```
Instead of having this here, line[8] you could have placed it in ```def setUp``` method.

mutebironald replied:
yes, point noted.

matthewwacha commented on hc/api/tests/test_create_check.py:
```diff
@@ -17,7 +17,8 @@ def post(self, data, expected_error=None):
             ### Assert that the expected error is the response error
-
+            result = r.json()
+            self.assertEqual(result['error'], expected_error)
```
Plus one on this line

matthewwacha commented on hc/api/tests/test_create_check.py:
```diff
@@ -44,26 +45,39 @@ def test_it_works(self):
         r = {'status_code': 201} ### This is just a placeholder variable
 
-        self.assertEqual(r['status_code'], 201)
+        r = self.client.post(self.URL, payload,
```
give the response variable a different name line[57]

mutebironald replied:
I think similar naming is not really an issue because the second one overwrites the first.

matthewwacha commented on hc/api/tests/test_create_check.py:
```diff
@@ -44,26 +45,39 @@ def test_it_works(self):
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "wrong api_key")
+
+        r = self.client.post(self.URL, payload,
```
rename variable

EdwinKato replied:
Also use this as an opportunity to run away from the single letter naming and go for something like `response`

EdwinKato committed:
Merge pull request #3 from andela/ch-heroku-deployment-158549842

chore(heroku-deployment)Deploy the application to heroku

EdwinKato commented on hc/api/tests/test_admin.py:
```diff
@@ -17,3 +20,4 @@ def test_it_shows_channel_list_with_pushbullet(self):
         ch.save()
 
         ### Assert for the push bullet
+        self.assertEqual(ch.kind, "pushbullet")
```
Checking for the kind in the `ch` is not a conclusive test. The idea is that a channel is added in the lines above, so you have to check that the overall channels have the new channel with the kind `pushbullet`

EdwinKato replied:
@mutebironald Lines 19 and 20 create a channel and save it. Check in the list of saved channels for one that has `pushbullet`

EdwinKato commented on hc/api/tests/test_create_check.py:
```diff
@@ -44,26 +45,39 @@ def test_it_works(self):
         self.assertEqual(check.tags, "bar,baz")
         self.assertEqual(check.timeout.total_seconds(), 3600)
         self.assertEqual(check.grace.total_seconds(), 60)
+        self.assertEqual(check.last_ping,None)
```
Leave a space after the comma

EdwinKato commented on hc/api/tests/test_create_check.py:
```diff
@@ -78,4 +92,13 @@ def test_it_rejects_non_string_name(self):
                   expected_error="name is not a string")
 
     ### Test for the assignment of channels
```
Missing test here

EdwinKato commented:
Nice start, work on the conflicts and the requested changes

JaymesKat commented on hc/api/tests/test_create_check.py:
```diff
@@ -44,26 +45,39 @@ def test_it_works(self):
+        payload = json.dumps({"name": "Foo", "api_key": "efg"})
+        
         ### Make the post request with a missing body and get the response
         r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
```
This line is longer than the recommended pep8 standard of 75 characters. The comment here may not be necessary

magicmarie committed:
This commit fixes the test errors in hc.api.tests from PR comments

Fixes tests for set Alice as staff and superuser, assign all channels,
Renames variables,  check for kind pushbullet in the channel objects

[Delivers #158549842]

magicmarie committed:
This commit fixes the test errors in hc.api.tests from PR comments

Fixes tests for set Alice as staff and superuser, assign all channels,
Renames variables,  check for kind pushbullet in the channel objects

[Delivers #158549842]

magicmarie committed:
Update the requirements.txt
